### PR TITLE
Small DX improvements for entityTransformation

### PR DESCRIPTION
**Pull request by alexonaci**
Hi @Charlesthebird , thank you for helping with entityTransformation, I tested the functionality and added here some very small suggestions/improvements. Please let me know what you think:)

- export API and ApiVersion types to be reachable to client application when used in entityTransformation  such that consumer applications wouldn' need to use `any` to deactivate type checking
- update README and index.ts example with examples that has type checking and don't use any
- update example to not use param reassigning - it might lead to unwanted side-effects inside the library due to reference change and is usually not a good practice (https://eslint.org/docs/latest/rules/no-param-reassign)

### CODE REVIEW

File: packages/backend/src/index.ts [hunk 1]
```diff
@@ -14,20 +14,19 @@ import {
...
+const entityTransformation = async (entity: Entity, api: ApiVersion | API) => {
   // The following commented out lines would add an "example-" prefix to your Entities.
-  // entity = {
+  // return {
   //   ...entity,
   //   metadata: {
   //     ...entity.metadata,
   //     title: 'example-' + (entity?.metadata?.title ?? ''),
   //   },
   // };
```
Comment (Charlesthebird) [inline packages/backend/src/index.ts]:
```suggestion
  // };
  return entity;
```

File: plugins/platform-portal-backstage-plugin-backend/README.md [hunk 2]
```diff
@@ -33,21 +33,20 @@ import {
...
+const entityTransformation = async (entity: Entity, api: ApiVersion | API) => {
   // The following commented out lines would add an "example-" prefix to your Entities.
-  // entity = {
+  // return {
   //   ...entity,
   //   metadata: {
   //     ...entity.metadata,
   //     title: 'example-' + (entity?.metadata?.title ?? ''),
   //   },
   // };
```
Comment (Charlesthebird) [inline plugins/platform-portal-backstage-plugin-backend/README.md]:
```suggestion
  // };
  return entity;
```

**Review by Charlesthebird — CHANGES_REQUESTED** [review]
@alexonaci Thank you for the type updates! Once these 2 changes are accepted I can approve and publish a new release.

File: packages/backend/src/index.ts [hunk 3]
```diff
@@ -14,20 +14,21 @@ import {
...
   return entity;
+  return entity;
```
Comment (Charlesthebird) [inline packages/backend/src/index.ts]:
```suggestion
```

Comment (alexonaci) [reply packages/backend/src/index.ts]: ouch, thanks for the catch:D

**Review by Charlesthebird — APPROVED** [review]

**Comment by Charlesthebird** [comment]
This is now in [v0.0.35](https://www.npmjs.com/package/@solo.io/platform-portal-backstage-plugin-backend) 🎉 !

**Pull request merged**